lib/mix/lib/mix/utils.ex

```diff
@@ -464,16 +464,30 @@ defmodule Mix.Utils do
   defp proxy_env do
     http_proxy  = System.get_env("HTTP_PROXY")  || System.get_env("http_proxy")
     https_proxy = System.get_env("HTTPS_PROXY") || System.get_env("https_proxy")
+    no_proxy    = System.get_env("NO_PROXY")    || System.get_env("no_proxy")
+                |> no_proxy_list()
 
-    {proxy_setup(:http, http_proxy), proxy_setup(:https, https_proxy)}
+    {proxy_setup(:http, http_proxy, no_proxy), proxy_setup(:https, https_proxy, no_proxy)}
   end
 
-  defp proxy_setup(scheme, proxy) do
+  defp no_proxy_list(nil) do
+    []
+  end
+
+  defp no_proxy_list(no_proxy_str) do
+    
+    no_proxy_list = String.split(no_proxy_str, ",")
+    for item <- no_proxy_list do
+      to_charlist(item)
+    end
```

Comment:
Maybe we could refactor all this to be just 

```elixir
Enum.map(String.split(no_proxy_str, ","), &to_charlist/1)
```

or maybe

```elixir
no_proxy_str
|> String.split(",")
|> Enum.map(&to_charlist/1)
```

if we want to be less concise. Wdyt? :)